### PR TITLE
Implement site blocking

### LINKED ISSUE #17: Implement Site Blocking
- You can add/remove a list of sites to block.
- When your Pomodoro state is focus, you can't connect a list of blocked sites.
- When your Pomodoro is short or long break, you can connect a list of blocked sites.

### PR DESCRIPTION
**Pull request by utilForever**
Closes #17

## Summary

Implements the site blocking feature for focustime:

- **During Focus sessions**: blocked sites are written to `/etc/hosts`, preventing connections
- **During Break sessions**: entries are removed from `/etc/hosts`, restoring access
- **On app exit**: cleanup removes any remaining `/etc/hosts` entries

### New: `src/blocker.rs`
- `SiteBlocker` struct with a `Vec<String>` blocklist
- `block()` / `unblock()` – writes/removes a `# focustime-block-start … end` section in `/etc/hosts`
- `add_site` / `remove_site` helpers (normalises to lowercase, deduplicates)
- `cleanup()` – called on app exit as best-effort cleanup
- 9 unit tests covering parsing, CRUD, and edge cases

### Updated: `src/app.rs`
- New `AppMode { Timer, SiteManager }` enum
- `SiteBlocker` integrated into `App`; `apply_blocking_for_phase()` called on every phase transition (auto-advance, Space, `n`)
- SiteManager key bindings: `[b]` open, `[a]` add, `[d]`/Delete remove, `↑`/`↓` navigate, `Esc`/`b` back

### Updated: `src/ui.rs`
- New `render_site_manager()` with site list (stateful highlight), text input area, blocking status indicator, permission error display, and key hints
- Timer view hints updated to include `[b]`

### Updated: `src/main.rs`
- Added `mod blocker`
- Calls `app.blocker.cleanup()` on exit

## Notes

- `/etc/hosts` modification requires elevated privileges. If the write fails (e.g. not running with `sudo`), the error is displayed in the TUI rather than crashing.
- Site list is not persisted across sessions (MVP scope per the issue).

### CODE REVIEW

File: src/blocker.rs [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+    }
+
+    fn remove_hosts_block(&self) -> io::Result<()> {
+        let content = fs::read_to_string(HOSTS_FILE)?;
+        let cleaned = Self::strip_block_section(&content);
+        // Only write back if something was actually removed.
+        if cleaned != content {
+            fs::write(HOSTS_FILE, cleaned)?;
+            flush_dns_cache();
+        }
```
[the remark this answers is not in the document]
Comment (utilForever) [reply src/blocker.rs]: Fixed.

File: src/blocker.rs [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+}
+
+/// Write `content` to the hosts file atomically via a temp file + rename so
+/// an interrupted write cannot corrupt the file or leave it truncated.
+fn atomic_write_hosts(content: &str) -> io::Result<()> {
+    let hosts_path = Path::new(HOSTS_FILE);
+    let dir = hosts_path.parent().unwrap_or(Path::new("."));
+    let tmp_path = dir.join(".focustime_hosts.tmp");
+    fs::write(&tmp_path, content)?;
+    fs::rename(&tmp_path, hosts_path)
```
[the remark this answers is not in the document]
Comment (utilForever) [reply src/blocker.rs]: Fixed.
